src/daemon/distribution.c

```diff
@@ -0,0 +1,205 @@
+//
+// Created by Svetlana Shmidt on 24.07.2020.
+//
+
+#include "distribution.h"
+
+#include <errno.h>
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+
+typedef struct bucket_s {
+  uint64_t bucket_counter;
+  double minimum, maximum;
+} bucket_t;
+
+struct distribution_s {
+  bucket_t *tree;
+  size_t num_buckets;
+  double total_sum;
+};
+
+size_t left_child_index(size_t node_index, size_t left, size_t right) {
```

Comment:
The `left` and `right` arguments are only for consistency with the `right_child_index` function and are unused here. It is good practice to flag unused arguments, which is done with a GCC / Clang compile extension:

```c
static size_t left_child_index(size_t node_index,
                               __attribute__((unused)) size_t left,
                               __attribute__((unused)) size_t right) {
```